### PR TITLE
Char level ppl

### PR DESCRIPTION
**Pull request by sdtblck**
Sorry, my autoformatter went ham.
Probably not the worst thing. 

Relevant changes are:

- adds a char counter here: https://github.com/EleutherAI/gpt-neox/compare/char_level_ppl?expand=1#diff-b6ee2c1db780b46787ab3a576020f9a12695f3f5cb38ba8badcd5501960e5d22R440
- adds the option to turn on char level ppl here: https://github.com/EleutherAI/gpt-neox/compare/char_level_ppl?expand=1#diff-fefa29324180c69866ca857d29fb03443ef143b9fee5aa5217cd5f5e5ae2b82cR892
- In eval, wraps the data iterator in a char counter that counts the characters and tokens as we iterate through it (needed to do it this way because we don't have access to the deepspeed internals when doing pipe parallel), then uses this info to calculate the char level ppl here: https://github.com/EleutherAI/gpt-neox/compare/char_level_ppl?expand=1#diff-e2b248f8c422a601bcb0b7d93f96c1dff070f2694737e2b69f1def64ab9c1844R662

### CODE REVIEW

File: megatron/neox_arguments/neox_args.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ class NeoXArgsModel(NeoXArgsTemplate):
     Scalenorm epsilon
     """
 
-    pos_emb: Literal["learned", "rotary", "sinusoidal", "rpe", "none"] = "learned"
+    pos_emb: Literal['learned', 'rotary', 'sinusoidal', 'rpe', 'alibi', 'none'] = "learned"
```
[the remark this answers is not in the document]
Comment (sdtblck) [reply megatron/neox_arguments/neox_args.py]: My autoformatter that mangled this PR is black, i like it.

[the remark this answers is not in the document]
Comment (StellaAthena) [reply megatron/neox_arguments/neox_args.py]: This PR also changes lines in `megatron/neox_arguments/neox_args.py` from single to double quotes. Which way do you want to be the standard?

**Review by sweinbach — DISMISSED** [review]

File: megatron/neox_arguments/neox_args.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sdtblck) [reply megatron/neox_arguments/neox_args.py]: Whatever my formatter says (so double :laughing: )

[the remark this answers is not in the document]
Comment (StellaAthena) [reply megatron/neox_arguments/neox_args.py]: Noted for the future

**Comment by StellaAthena** [comment]
I have resolved the conflicts with main. @sweinbach can you confirm that this doesn't introduce bugs for whatever reason?

**Comment by sdtblck** [comment]
@StellaAthena I tested again, works fine. Can you reapprove?

**Review by EricHallahan — APPROVED** [review]

**Pull request merged**